### PR TITLE
[zh]Add 2022-05-27-maxunavailable-for-statefulset.md chinese version

### PR DESCRIPTION
**Pull request by zhangxyjlu**
fix:#34035
Add 2022-05-27-maxunavailable-for-statefulset.md chinese version

### CODE REVIEW

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+Here are some examples:
+
+-  I am using a StatefulSet to orchestrate a multi-instance, cache based application where the size of the cache is large. The cache 
+   starts cold and requires some siginificant amount of time before the container can start. There could be more initial startup tasks
+   that are required. A RollingUpdate on this StatefulSet would take a lot of time before the application is fully updated. If the 
+   StatefulSet supported updating more than one pod at a time, it would result in a much faster update.
+-->
+这里有些例子：
+
+- 我使用 StatefulSet 来编排一个基于缓存的多实例应用程序，其中缓存的大小很大。
```
Comment (howieyuen) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:42]:
```suggestion
- 我使用 StatefulSet 来编排一个基于缓存的多实例应用程序，其中缓存的规格很大。
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+这是 StatefulSet `spec` 的一部分。例如：
+
+```
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: web
+  namespace: default
+spec:
+  podManagementPolicy: OrderedReady  # you must set OrderedReady
```
Comment (howieyuen) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:74]: 对于代码片段的注释，我们一般不保留原文做对照，直接翻译；下同
```suggestion
  podManagementPolicy: OrderedReady  # 你必须设为 OrderedReady
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: k8s.gcr.io/nginx-slim:0.8
+        imagePullPolicy: IfNotPresent
+        name: nginx
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 2 # this is the new alpha field, whose default value is 1
```
Comment (howieyuen) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:90]:
```suggestion
      maxUnavailable: 2 # 这是 alpha 特性的字段，默认值是 1
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+那么不仅是 `maxUnavailable` 数量的副本同时被终止，还会导致 `maxUnavailable` 数量的副本同时在
+`ContainerCreating` 阶段。这就是所谓的 bursting。
+
+<!--
+So, now you may have a lot of questions about:-
+- What is the behavior when you set `podManagementPolicy: Parallel`?
+- What is the behavior when `partition` to a value other than `0`?
+-->
+因此，现在你可能有很多关于以下方面的问题：
+- 当设置`podManagementPolicy:Parallel` 时，会产生什么行为？
```
Comment (howieyuen) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:215]:
```suggestion
- 当设置 `podManagementPolicy:Parallel` 时，会产生什么行为？
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+<!--
+## Further reading and next steps {#next-steps}
+- [Maximum unavailable Pods](/docs/concepts/workloads/controllers/statefulset/#maximum-unavailable-pods)
+- [KEP for MaxUnavailable for StatefulSet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/961-maxunavailable-for-statefulset)
+- [Implementation](https://github.com/kubernetes/kubernetes/pull/82162/files)
+- [Enhancement Tracking Issue](https://github.com/kubernetes/enhancements/issues/961)
+-->
+## 进一步阅读和后续步骤 {#next-steps}
+- [最大不可用 Pod](/zh/docs/concepts/workloads/controllers/statefulset/#maximum-unavailable-pods)
```
Comment (howieyuen) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:236]:
```suggestion
- [最多不可用 Pod 数](/zh/docs/concepts/workloads/controllers/statefulset/#maximum-unavailable-pods)
```

**Review by howieyuen — CHANGES_REQUESTED** [review]

**Comment by howieyuen** [comment]
Cool, thanks for your contribution
/approve

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+并且如果有这种情况的话，你必须确保你的应用程序能够处理发生的此类顺序问题。当你设置 `maxUnavailable`
+值大于 1 时，更新 Pod 的批次之间会保证顺序。该保证意味着在批次 0（副本 4 和 3）中的 Pod
+准备好之前，更新批次 2（副本 2 和 1）中的 Pod 无法开始更新。
+
+<!--
+Although Kubernetes refers to these as _replicas_, your stateful application may have a different view and each pod of the StatefulSet may
+be holding completely different data than other pods. The important thing here is that updates to StatefulSets happen in batches, and you can
+now have a batch size larger than 1 (as an alpha feature).
+-->
+尽管 Kubernetes 将这些称为**副本**，但你的有状态应用程序可能具有不同的视图，并且 StatefulSet 的每个
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:195]:
```suggestion
尽管 Kubernetes 将这些称为**副本**，但你的有状态应用程序可能不这样理解，StatefulSet 的每个
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+Pod 可能持有与其他 Pod 完全不同的数据。重要的是，StatefulSet 的更新是分批进行的，
+你现在可以拥有大于 1 的批大小（作为 alpha 特性）。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:197]:
```suggestion
你现在让批次大小大于 1（作为 alpha 特性）。
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+<!--
+Also note, that the above behavior is with `podManagementPolicy: OrderedReady`. If you defined a StatefulSet as `podManagementPolicy: Parallel`,
+not only `maxUnavailable` number of replicas are terminated at the same time; `maxUnavailable` number of replicas start in `ContainerCreating`
+phase at the same time as well. This is called bursting.
+-->
+还要注意，上面的行为采用的 Pod 管理策略是 `podManagementPolicy: OrderedReady`。
+如果你的 StatefulSet 的 Pod 管理策略是 `podManagementPolicy: Parallel`，
+那么不仅是 `maxUnavailable` 数量的副本同时被终止，还会导致 `maxUnavailable` 数量的副本同时在
+`ContainerCreating` 阶段。这就是所谓的 bursting。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:207]:
```suggestion
`ContainerCreating` 阶段。这就是所谓的突发（Bursting）。
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+
+<!--
+So, now you may have a lot of questions about:-
+- What is the behavior when you set `podManagementPolicy: Parallel`?
+- What is the behavior when `partition` to a value other than `0`?
+-->
+因此，现在你可能有很多关于以下方面的问题：
+- 当设置 `podManagementPolicy:Parallel` 时，会产生什么行为？
+- 将 `partition` 设置为 `0` 以外的值时会发生什么？
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:216]:
```suggestion
- 将 `partition` 设置为非 `0` 值时会发生什么？
```

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+<!--
+It might be better to try and see it for yourself. This is an alpha feature, and the Kubernetes contributors are looking for feedback on this feature. Did
+this help you achieve your stateful scenarios Did you find a bug or do you think the behavior as implemented is not intuitive or can
+break applications or catch them by surprise? Please [open an issue](https://github.com/kubernetes/kubernetes/issues) to let us know.
+-->
+自己试试看可能会更好。这是一个 alpha 特性，Kubernetes 贡献者正在寻找有关此特性的反馈。
+这是否有助于你实现有状态的场景？你是否发现了一个 bug，或者你认为实现的行为不直观易懂，
+或者它可能会破坏应用程序或让他们感到吃惊？请[创建一个 issue](https://github.com/kubernetes/kubernetes/issues)
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:225]:
```suggestion
或者它可能会破坏应用程序或让他们感到吃惊？请[登记一个 issue](https://github.com/kubernetes/kubernetes/issues)
```

**Comment by tengqm** [comment]
/approve

It is desirable to mark the feedbacks as resolved if changes are made accordingly.

File: content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+web-0   0/1     ContainerCreating   0          0s
+web-0   1/1     Running             0          1s
+```
+<!--
+Note that as soon as the rolling update starts, both 4 and 3 (the two highest ordinal pods) start terminating at the same time. Pods 
+with ordinal 4 and 3 may become ready at their own pace. As soon as both pods 4 and 3 are ready, pods 2 and 1 start terminating at the
+same time. When pods 2 and 1 are both running and ready, pod 0 starts terminating. 
+-->
+注意，滚动更新一开始，4 和 3（两个最高序号的 Pod）同时开始进入 `Terminating` 状态。
+Pod 4 和 3 会按照自身节奏进行更新。一旦 Pod 4 和 3 更新完毕后， Pod 2 和 1 会同时进入
```
Comment (Sea-n) [inline content/zh/blog/_posts/2022-05-27-maxunavailable-for-statefulset.md:169]: nit: Redundant space here.
```suggestion
Pod 4 和 3 会按照自身节奏进行更新。一旦 Pod 4 和 3 更新完毕后，Pod 2 和 1 会同时进入
```

**Review by Sea-n — COMMENTED** [review]
Look good to me!
Maybe you would like to do a final review & adjustment.

**Comment by Sea-n** [comment]
Thanks for your hard work, I think we all agree satisfied with the translation!
/lgtm

**Pull request merged**